src/h5web/providers/hsds/api.ts

```diff
@@ -178,83 +143,121 @@ export class HsdsApi implements ProviderAPI {
     return Promise.all(attrsPromises);
   }
 
-  /* Processing methods to fetch links and attributes in addition to the entity. Also updates API members. */
-
   private async processGroup(
     id: HDF5Id,
-    path = '',
-    depth = Infinity
-  ): Promise<void> {
+    path: string,
+    name: string,
+    depth = 1
```

Comment:
I think `depth` should not have a default value. This way, we know exactly at each call how deep the processing will go.